app/src/main/java/com/example/encoratask/ItemListActivity.kt

```diff
@@ -23,70 +22,48 @@ import com.example.encoratask.dummy.DummyContent
  * item details. On tablets, the activity presents the list of items and
  * item details side-by-side using two vertical panes.
  */
-class ItemListActivity : AppCompatActivity() {
+class ItemListActivity : AppCompatActivity(), View.OnClickListener {
+
+    var characters:ArrayList<Character>? = null
 
     /**
      * Whether or not the activity is in two-pane mode, i.e. running on a tablet
      * device.
      */
 
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_item_list)
-
-        val toolbar = findViewById<Toolbar>(R.id.toolbar)
-        setSupportActionBar(toolbar)
-        toolbar.title = title
-
-        findViewById<FloatingActionButton>(R.id.fab).setOnClickListener { view ->
-            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                    .setAction("Action", null).show()
-        }
-
-        setupRecyclerView(findViewById(R.id.item_list))
+        this.getCharacters()
     }
 
-    private fun setupRecyclerView(recyclerView: RecyclerView) {
-        recyclerView.adapter = SimpleItemRecyclerViewAdapter(DummyContent.ITEMS)
-    }
-
-    class SimpleItemRecyclerViewAdapter(private val values: List<DummyContent.DummyItem>) :
-            RecyclerView.Adapter<SimpleItemRecyclerViewAdapter.ViewHolder>() {
-
-        private val onClickListener: View.OnClickListener
-
-        init {
-            onClickListener = View.OnClickListener { v ->
-                val item = v.tag as DummyContent.DummyItem
-                val intent = Intent(v.context, ItemDetailActivity::class.java).apply {
-                    putExtra(ItemDetailFragment.ARG_ITEM_ID, item.id)
-                }
-                v.context.startActivity(intent)
-            }
-        }
+    private fun getCharacters() {
 
-        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-            val view = LayoutInflater.from(parent.context)
-                    .inflate(R.layout.item_list_content, parent, false)
-            return ViewHolder(view)
-        }
+        CharacterService.getCharacters(this, {response ->
+            characters = Character.parse(response)
+            setupRecyclerView()
+        },
+        {error ->
+            Toast.makeText(this, error.localizedMessage, Toast.LENGTH_LONG).show()
```

Comment:
This line will show an empty screen with the Toast with information that user should not know (localizedMessage). The user should not know headers, response from server or classes in your project

Reply:
I understand, but for example porpuses I think its ok